src/updaters/dotnet/common-properties.ts

```diff
@@ -0,0 +1,44 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {checkpoint, CheckpointType} from '../../util/checkpoint';
+import {GitHubFileContents} from '../../github';
+import {Update, UpdateOptions, VersionsMap} from '../update';
+
+export class CommonProperties implements Update {
+  path: string;
+  changelogEntry: string;
+  version: string;
+  versions?: VersionsMap;
+  packageName: string;
+  create: boolean;
+  contents?: GitHubFileContents;
+
+  constructor(options: UpdateOptions) {
+    this.create = false;
+    this.path = options.path;
+    this.changelogEntry = options.changelogEntry;
+    this.version = options.version;
+    this.packageName = options.packageName;
+    this.contents = options.contents;
+  }
+
+  updateContent(content: string): string {
+    checkpoint(`updating ${this.path}`, CheckpointType.Success);
```

Comment:
Is this doing a string replace on a string that contains json or xml?

Reply:
Doing a string replace on structured data is fragile, but sometimes the most feasible option.

How difficult would it be to parse the XML, update the element, and then re-serialize the XML?